Tests on GPU failing

## Describe the bug
<!-- A clear and concise description of what the bug is. -->
Tests in test/test_gpu.py do not all pass (test_gpu_sinkhorn). 
There is a discrepancy between the matrix computed by ot.bregman and the one computed by ot.gpu.bregman. 

### To Reproduce
Steps to reproduce the behavior:
1. Run test/test_gpu.py

<!-- If you have error messages or stack traces, please provide it here as well -->
```
=================================== FAILURES ===================================
______________________________ test_gpu_sinkhorn _______________________________

    @pytest.mark.skipif(nogpu, reason="No GPU available")
    def test_gpu_sinkhorn():
    
        rng = np.random.RandomState(0)
    
        for n_samples in [50, 100, 500, 1000]:
            a = rng.rand(n_samples // 4, 100)
            b = rng.rand(n_samples, 100)
    
            wa = ot.unif(n_samples // 4)
            wb = ot.unif(n_samples)
    
            wb2 = np.random.rand(n_samples, 20)
            wb2 /= wb2.sum(0, keepdims=True)
    
            M = ot.dist(a.copy(), b.copy())
            M2 = ot.gpu.dist(a.copy(), b.copy(), to_numpy=False)
    
            reg = 1
    
            G = ot.sinkhorn(wa, wb, M, reg)
            G1 = ot.gpu.sinkhorn(wa, wb, M, reg)

>           np.testing.assert_allclose(G1, G, rtol=1e-10)
E           AssertionError: 
E           Not equal to tolerance rtol=1e-10, atol=0
E           
E           Mismatched elements: 600 / 600 (100%)
E           Max absolute difference: 1.37138433e-07
E           Max relative difference: 2.00548806e-05
E            x: array([[5.888717e-04, 8.415569e-05, 8.951892e-05, 2.190684e-03,
E                   1.977490e-05, 9.029307e-03, 2.036359e-04, 2.300168e-03,
E                   2.933039e-04, 1.124758e-04, 3.394681e-04, 1.416449e-03,...
E            y: array([[5.888707e-04, 8.415610e-05, 8.951908e-05, 2.190691e-03,
E                   1.977479e-05, 9.029328e-03, 2.036355e-04, 2.300179e-03,
E                   2.933086e-04, 1.124764e-04, 3.394675e-04, 1.416444e-03,...

test/test_gpu.py:73: AssertionError
```

#### Screenshots
<!-- If applicable, add screenshots to help explain your problem. -->

#### Code sample
<!-- Ideally attach a minimal code sample to reproduce the decried issue. 
Minimal means having the shortest code but still preserving the bug. -->
```
from test import test_gpu
test_gpu.test_gpu_sinkhorn()
```

### Expected behavior
<!-- A clear and concise description of what you expected to happen. -->
GPU unit tests should pass.

### Environment (please complete the following information):
- OS (e.g. MacOS, Windows, Linux): Linux
- Python version: 3.8
- How was POT installed (source, `pip`, `conda`): source
- Build command you used (if compiling from source): make buildext ; make install
- Only for GPU related bugs:
    - CUDA version: 10.2
    - GPU models and configuration: Happens on NVIDIA V100 and NVIDIA A100
    - Any other relevant information: CuPY version is 9.0.0

Output of the following code snippet:
```python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import ot; print("POT", ot.__version__)
```
```
Linux-4.18.0-147.48.1.el8_1.x86_64-x86_64-with-redhat-8.1-Ootpa
Python 3.7.11 (default, Jul 27 2021, 14:32:16) 
[GCC 7.5.0]
NumPy 1.21.2
SciPy 1.7.1
POT 0.8.0dev
```

### Additional context
<!-- Add any other context about the problem here. -->

--- a/ot/bregman.py
+++ b/ot/bregman.py
@@ -312,7 +312,7 @@ def sinkhorn_knopp(a, b, M, reg, numItermax=1000,
     numItermax : int, optional
         Max number of iterations
     stopThr : float, optional
-        Stop threshol on error (>0)
+        Stop threshold on error (>0)
     verbose : bool, optional
         Print information along iterations
     log : bool, optional

--- a/ot/gpu/bregman.py
+++ b/ot/gpu/bregman.py
@@ -154,7 +154,7 @@ def sinkhorn_knopp(a, b, M, reg, numItermax=1000, stopThr=1e-9,
                 # compute right marginal tmp2= (diag(u)Kdiag(v))^T1
                 tmp2 = np.sum(u[:, None] * K * v[None, :], 0)
                 #tmp2=np.einsum('i,ij,j->j', u, K, v)
-                err = np.linalg.norm(tmp2 - b) # violation of marginal
+                err = np.linalg.norm(tmp2 - b)  # violation of marginal
             if log:
                 log['err'].append(err)
 

--- a/ot/gpu/bregman.py
+++ b/ot/gpu/bregman.py
@@ -148,8 +148,10 @@ def sinkhorn_knopp(a, b, M, reg, numItermax=1000, stopThr=1e-9,
             # we can speed up the process by checking for the error only all
             # the 10th iterations
             if nbb:
-                err = np.sum((u - uprev)**2) / np.sum((u)**2) + \
-                    np.sum((v - vprev)**2) / np.sum((v)**2)
+                err = np.sqrt(
+                    np.sum((u - uprev)**2) / np.sum((u)**2)
+                    + np.sum((v - vprev)**2) / np.sum((v)**2)
+                )
             else:
                 # compute right marginal tmp2= (diag(u)Kdiag(v))^T1
                 tmp2 = np.sum(u[:, None] * K * v[None, :], 0)

--- a/ot/gpu/bregman.py
+++ b/ot/gpu/bregman.py
@@ -54,7 +54,7 @@ def sinkhorn_knopp(a, b, M, reg, numItermax=1000, stopThr=1e-9,
     numItermax : int, optional
         Max number of iterations
     stopThr : float, optional
-        Stop threshol on error (>0)
+        Stop threshold on error (>0)
     verbose : bool, optional
         Print information along iterations
     log : bool, optional
@@ -154,7 +154,7 @@ def sinkhorn_knopp(a, b, M, reg, numItermax=1000, stopThr=1e-9,
                 # compute right marginal tmp2= (diag(u)Kdiag(v))^T1
                 tmp2 = np.sum(u[:, None] * K * v[None, :], 0)
                 #tmp2=np.einsum('i,ij,j->j', u, K, v)
-                err = np.linalg.norm(tmp2 - b)**2  # violation of marginal
+                err = np.linalg.norm(tmp2 - b) # violation of marginal
             if log:
                 log['err'].append(err)
 

--- a/ot/gpu/da.py
+++ b/ot/gpu/da.py
@@ -120,7 +120,7 @@ def sinkhorn_lpl1_mm(a, labels_a, b, M, reg, eta=0.1, numItermax=10,
     labels_a2 = cp.asnumpy(labels_a)
     classes = npp.unique(labels_a2)
     for c in classes:
-        idxc, = utils.to_gpu(npp.where(labels_a2 == c))
+        idxc = utils.to_gpu(*npp.where(labels_a2 == c))
         indices_labels.append(idxc)
 
     W = np.zeros(M.shape)

